Template/HiP_MicroServiceTemplate/Controllers/FooController.cs

```diff
@@ -79,18 +79,30 @@ public IActionResult GetById(int id)
             if (!ModelState.IsValid)
                 return BadRequest(ModelState);
 
-            var ev = new FooCreated
-            {
-                Id = _entityIndex.NextId(ResourceType.Foo),
-                UserId = User.Identity.GetUserIdentity(),
-                Properties = args,
-                Timestamp = DateTimeOffset.Now
-            };
+            var id = _entityIndex.NextId(ResourceTypes.Foo);
+            await EntityManager.CreateEntityAsync(_eventStore, args, ResourceTypes.Foo, id, User.Identity.GetUserIdentity());
+            return Created($"{Request.Scheme}://{Request.Host}/api/Foo/{id}", id);
+        }
+
+        [HttpPut("{id}")]
+        [ProducesResponseType(404)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(204)]
 
-            await _eventStore.AppendEventAsync(ev);
-            return Created($"{Request.Scheme}://{Request.Host}/api/Foo/{ev.Id}", ev.Id);
+        public async Task<IActionResult> PutAsync(int id, [FromBody]FooArgs args)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            if (!_entityIndex.Exists(ResourceTypes.Foo, id))
+                return NotFound();
+            var oldFoo = await EventStreamExtensions.GetCurrentEntityAsync<FooArgs>(_eventStore.EventStream, ResourceTypes.Foo, id);
```

Comment:
`GetCurrentEntityAsync` is an extension method - better use it via `_eventStore.EventStream.GetCurrentEntityAsync(...)`